src/cloud-sql-instance.ts

```diff
@@ -107,51 +115,118 @@ export class CloudSQLInstance {
   }
 
   async refresh(): Promise<void> {
+    const currentRefreshId = this.scheduledRefreshID;
+
     // Since forceRefresh might be invoked during an ongoing refresh
     // we keep track of the ongoing promise in order to be able to await
     // for it in the forceRefresh method.
     // In case the throttle mechanism is already initialized, we add the
     // extra wait time `limitRateInterval` in order to limit the rate of
     // requests to Cloud SQL Admin APIs.
-    this.ongoingRefreshPromise = this.throttle
-      ? this.throttle(this._refresh).call(this)
-      : this._refresh();
-
-    // awaits for the ongoing promise to resolve, since the refresh is
-    // completed once the promise is resolved, we just free up the reference
-    // to the promise at this point, ensuring any new call to `forceRefresh`
-    // is able to trigger a new refresh
-    await this.ongoingRefreshPromise;
-    this.ongoingRefreshPromise = undefined;
+    this.ongoingRefreshPromise = (
+      this.throttle && this.scheduledRefreshID
+        ? this.throttle(this.refreshValues).call(this)
+        : this.refreshValues()
+    )
+      // These needs to be part of the chain of promise referenced in
+      // ongoingRefreshPromise in order to avoid race conditions
+      .then((nextValues: RefreshableValues) => {
+        // in case the id at the moment of starting this refresh cycle has
+        // changed, that means that it has been canceled
+        if (currentRefreshId !== this.scheduledRefreshID) {
+          return nextValues;
```

Comment:
`nextValues` is a confusing name to me. In this case `nextValues` is returning the old values right?

Reply:
oooohhh, good catch, that was just a smell. this function doesn't need to return any value! updating...